### ISSUE TITLE
Swipe to close/close on drag

### ISSUE THREAD

**Issue by mostr** [root-two/react-native-drawer#138]
Is it possible to configure this component to allow closing drawer also by dragging/swiping it besides tapping outside? What I'd like to achieve is that I can show/hide drawer using swipe gestures only (say drawer is on the left -> I can swipe from left edge to the center to show it and then swipe from any place in left direction to close it)

**Comment by rgoldiez** [comment]
I'm seeing the same issue when moving from 1.x to 2.0 ... that is, you can't drag the drawer closed when `tapToClose={true}`

**Comment by mostr** [comment]
Haven't used 1.x version, it used to work there?

**Comment by rgoldiez** [comment]
Correct.... I have a drawer on the left side, and you can both swipe to close or tap the small visible portion of the main view to close the drawer.  Doesn't seem to work the same in 2.0

**Comment by mostr** [comment]
@rgoldiez could you share which specific version from 1.x you use and what's your component declaration (props)? Can't get it working even with some of the latest from 1.x.

Just to be sure, the feature in question is to be able to drag the drawer itself to close it (not only the visible part of main view).

**Comment by rgoldiez** [comment]
Sure... I'm using 1.13.0 with the following setup but also the code in this PR (that was not accepted as I wrote it): #71  

Let me know if you have any questions...

```
<Drawer
   ref='leftDrawer'
   content={<LeftDrawer />}
   type='static'
   side='left'
   openDrawerOffset={0.2}
   onOpen={this.openLeftDrawer}
   onOpenStart={() => {dismissKeyboard; StatusBarIOS.setHidden(true, 'slide')}}
   onClose={this.closeDrawers}
   styles={{main: {shadowColor: '#000000', shadowOpacity: 0.4, shadowRadius: 3}}}
   tweenHandler={Drawer.tweenPresets.parallax}
   tweenDuration={150}
   negotiatePan={true}
   panOpenMask={5}
   captureGestures={true}
   tapToClose={true}
   initializeOpen={false}
   panCloseMask={0.2}
   disabled={this.state.leftDrawerDisabled}
   >
   <View style={styles.container}>
      {/* ... main content ... */}
   </View>
</Drawer>
```

**Comment by rt2zz** [comment]
I don't have a simple answer here, this functionality was superceded by captureGesture/negotiatePan improvements. I think this might still be possible to support if we move processTapGestures from on press down to on press up.

**Comment by rgoldiez** [comment]
That makes sense based in the behavior in 1.x compared to 2.x.... In 1.x the tapToClose happens on the lift of the finger. In 2.0 it happens on the press down, so there's no opportunity to drag it. 

I prefer the tap happening on the up, not down.

**Comment by rgoldiez** [comment]
I just upgraded from `1.13.0` -> `2.1.0` and I can now mirror the user behavior I had before.  Couple small changes that I had to make, relative to my implementation above:
1. Changed from `var Drawer = requires('react-native-drawr');` to `import Drawer from react-native-drawer`
2. Changed the `panOpenMask={5}` prop -> `panOpenMask={0.95}` because it now wants a percentage of the screen width and no longer likes whole number pixels.
3. Changed `captureGestures={true}` prop -> `captureGestures={open}` so that tapping on the main view, including the icon to open the drawer, worked.

Pretty straightforward upgrade and `2.1.0` seems ready for production in my app.
